omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskDefinitionController1_8Test.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+import java.util.List;
+
+public class TaskDefinitionController1_8Test extends MainResourceControllerTest {
+	
+	@Autowired
+	RestService restService;
+	
+	private TaskDefinition testTask = new TaskDefinition(1, "TestTask", "TestTask Description",
+	        "org.openmrs.scheduler.tasks.TestTask");
+	
+	private MockTaskServiceWrapper mockTaskServiceWrapper = new MockTaskServiceWrapper();
+	
+	@Before
+	public void setUp() throws Exception {
+		testTask.setRepeatInterval(10L);
+		testTask.setStartOnStartup(true);
+		testTask.setProperty("propertyKey", "propertyValue");
+		
+		mockTaskServiceWrapper.registeredTasks.add(testTask);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	@Override
+	public void shouldGetAll() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		SimpleObject resultTasks = deserialize(handle(req));
+		List<Object> results = Util.getResultsList(resultTasks);
+		Assert.assertNotNull(resultTasks);
+		Assert.assertEquals(results.size(), getAllScheduledCount());
+	}
+	
+	@Test
+	public void shouldGetAllRegistered() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "registered");
+		SimpleObject resultTasks = deserialize(handle(req));
+		List<Object> results = Util.getResultsList(resultTasks);
+		Assert.assertNotNull(resultTasks);
+		Assert.assertEquals(results.size(), getAllCount());
+	}
+	
+	@Test
+	public void shouldGetModuleByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(result.get("name"));
+		Assert.assertNotNull(result.get("description"));
+		Assert.assertNotNull(result.get("taskClass"));
+	}
+	
+	@Test
+	@Override
+	public void shouldGetFullByUuid() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		req.addParameter("v", "full");
+		
+		SimpleObject result = deserialize(handle(req));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "name"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "description"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "taskClass"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "repeatInterval"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "startOnStartup"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "started"));
+		Assert.assertNotNull(PropertyUtils.getProperty(result, "properties"));
+	}
+	
+	@Test
+	@Override
+	public void shouldGetDefaultByUuid() throws Exception {
```

Comment:
When you say shouldGet, add what exactly you are getting.

Reply:
Here I have @Override the method from the super class. that will express the Default representation and I can't change the name here